src/include/detail/linalg/tdb_matrix_with_ids.h

```diff
@@ -224,19 +224,33 @@ class tdbBlockedMatrixWithIds
 
     auto layout_order = ids_schema_.cell_order();
     this->ids().resize(elements_to_load * dimension);
-    // Create a query
+
+    // Read TileDB data
+    size_t total_size = elements_to_load * dimension;
+    size_t offset = 0;
+    auto ptr = this->ids().data();
     tiledb::Query query(this->ctx_, *ids_array_);
-    query.set_subarray(subarray)
-        .set_layout(layout_order)
-        .set_data_buffer(attr_name, this->ids());
-    tiledb_helpers::submit_query(tdb_func__, ids_uri_, query);
+    query.set_subarray(subarray).set_layout(layout_order);
+    tiledb::Query::Status status;
+    do {
+      // Submit query and get status
+      size_t request_size = READ_BATCH_SIZE;
+      if (offset + READ_BATCH_SIZE > total_size) {
+        request_size = total_size - offset;
+      }
+      query.set_data_buffer(attr_name, ptr + offset, request_size);
```

Comment:
Should there be a check for ensuring we don't extend past the size of the data buffer?

Reply:
Restructured the loop and error handling. Let me know if this addresses your comments